onnxruntime/contrib_ops/cpu/quantization/quant_gemm.cc

```diff
@@ -47,14 +47,14 @@ class QGemm : protected GemmBase, public MatMulIntegerBase {
     bool a_is_signed = a->IsDataType<int8_t>();
     const uint8_t* a_data = static_cast<const uint8_t*>(a->DataRaw());
 
-    std::unique_ptr<Tensor> a_trans_buffer;
```

Comment:
This is what [yuslepukhin](https://github.com/yuslepukhin) suggested in another PR. Using std::optional to replace std::unique_ptr to reduce dynamical memory allocation.

Reply:
Yes, because it is in the inference path.